genet/schedule_elements.py

```diff
@@ -26,25 +35,58 @@ class ScheduleElement:
     Base class for Route, Service and Schedule
     """
 
-    def __init__(self, stops):
-        self._graph = self._build_graph(stops)
-        self.reference_nodes = list(self._graph.nodes())
-        self.reference_edges = list(self._graph.edges())
-        self.epsg = self.find_epsg()
-        self._graph.graph['crs'] = {'init': self.epsg}
+    def __init__(self):
+        # check if in graph first
+        if 'crs' in self._graph.graph:
+            self.epsg = self._graph.graph['crs']['init']
+        else:
+            self.epsg = self.find_epsg()
+            self._graph.graph['crs'] = {'init': self.epsg}
+
+    def _surrender_to_graph(self):
+        d = deepcopy(self.__dict__)
+        return d
+
+    def _get_service_from_graph(self, service_id):
+        if service_id in self._graph.graph['services']:
+            return Service(_graph=self._graph, **self._graph.graph['services'][service_id])
+        else:
+            raise ServiceIndexError(f'Service with index {service_id} not found')
+
+    def _get_route_from_graph(self, route_id):
+        if route_id in self._graph.graph['routes']:
+            return Route(_graph=self._graph, **self._graph.graph['routes'][route_id])
+        else:
+            raise RouteIndexError(f'Route with index {route_id} not found')
+
+    def _stop_ids_in_graph(self, stop_ids: List[str]):
+        return set(stop_ids).issubset(set(self._graph.nodes))
+
+    def _route_ids_in_graph(self, route_ids: List[str]):
+        return set(route_ids).issubset(set(self._graph.graph['routes'].keys()))
 
-    def _update_graph(self, new_graph):
-        self._graph = new_graph
+    def _service_ids_in_graph(self, service_ids: List[str]):
+        return set(service_ids).issubset(set(self._graph.graph['services'].keys()))
+
+    def change_log(self):
+        return self._graph.graph['change_log']
```

Comment:
added the changelog to the graph too, so it would be shared across/accessible to all objects within the Schedule